consul/consul.go

```diff
@@ -99,43 +145,26 @@ func (r *ConsulAdapter) buildCheck(service *bridge.Service) *consulapi.AgentServ
 		if timeout := service.Attrs["check_timeout"]; timeout != "" {
 			check.Timeout = timeout
 		}
-		if method := service.Attrs["check_http_method"]; method != "" {
-			check.Method = method
-		}
 	} else if path := service.Attrs["check_https"]; path != "" {
 		check.HTTP = fmt.Sprintf("https://%s:%d%s", service.IP, service.Port, path)
 		if timeout := service.Attrs["check_timeout"]; timeout != "" {
 			check.Timeout = timeout
 		}
-		if method := service.Attrs["check_https_method"]; method != "" {
-			check.Method = method
-		}
 	} else if cmd := service.Attrs["check_cmd"]; cmd != "" {
-		check.Args = []string{"check-cmd", service.Origin.ContainerID[:12], service.Origin.ExposedPort, cmd}
+		check.Script = fmt.Sprintf("check-cmd %s %s %s", service.Origin.ContainerID[:12], service.Origin.ExposedPort, cmd)
 	} else if script := service.Attrs["check_script"]; script != "" {
-		check.Args = []string{r.interpolateService(script, service)}
+		check.Script = r.interpolateService(script, service)
 	} else if ttl := service.Attrs["check_ttl"]; ttl != "" {
 		check.TTL = ttl
 	} else if tcp := service.Attrs["check_tcp"]; tcp != "" {
 		check.TCP = fmt.Sprintf("%s:%d", service.IP, service.Port)
 		if timeout := service.Attrs["check_timeout"]; timeout != "" {
 			check.Timeout = timeout
 		}
-	} else if grpc := service.Attrs["check_grpc"]; grpc != "" {
-		check.GRPC = fmt.Sprintf("%s:%d", service.IP, service.Port)
-		if timeout := service.Attrs["check_timeout"]; timeout != "" {
-			check.Timeout = timeout
-		}
-		if useTLS := service.Attrs["check_grpc_use_tls"]; useTLS != "" {
-			check.GRPCUseTLS = true
-			if tlsSkipVerify := service.Attrs["check_tls_skip_verify"]; tlsSkipVerify != "" {
-				check.TLSSkipVerify = true
-			}
-		}
 	} else {
 		return nil
 	}
-	if len(check.Args) != 0 || check.HTTP != "" || check.TCP != "" || check.GRPC != "" {
+	if check.Script != "" || check.HTTP != "" || check.TCP != "" {
```

Comment:
I guess we should keep this code here. Is latest features from registrator original upstream right? - Perhaps, this is showing as remove because you'll need to rebase your codebase with the our default `main` branch. Can you check this one?